.github/workflows/unit_tests.yml

```diff
@@ -16,36 +16,20 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.10", "3.11", "3.12", "3.13", "3.14"]
```

Comment:
If we stop testing Python 3.9 we should remember to update also this setting
https://github.com/campa-consortium/pals-python/blob/0356263f779a3fa86cc46f1cc9f098474a3c3133/pyproject.toml#L13
whenever we update the file for a new release.